--- Makefile.in
+++ Makefile.in
@@ -55,7 +55,7 @@
 ENABLE_LIB_STATIC = @ENABLE_LIB_STATIC@
 HAVE_WASI_SDK = @HAVE_WASI_SDK@
 
-CFLAGS = @CFLAGS@ @CPPFLAGS@
+CFLAGS = @CFLAGS@ @CPPFLAGS@ -D_REENTRANT -D_GNU_SOURCE
 #
 # $(LDFLAGS.configure) represents any LDFLAGS=... the client passes to
 # configure. See main.mk.
--- sqlite3.pc.in
+++ sqlite3.pc.in
@@ -10,4 +10,4 @@
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -lsqlite3
 Libs.private: @LDFLAGS_MATH@ @LDFLAGS_ZLIB@ @LDFLAGS_DLOPEN@ @LDFLAGS_PTHREAD@ @LDFLAGS_ICU@
-Cflags: -I${includedir}
+Cflags: -I${includedir} -D_REENTRANT -D_GNU_SOURCE
